### PR TITLE
Add memcpy check to CI

### LINKED ISSUE #334: Regression suite does not track memory movement

### PR DESCRIPTION
**Pull request by pgrete**
## PR Summary

We noticed in the past that sometimes mem copies between host and device are introduced in ways not always obvious, see also #293 

This tests now adds a check that for a uniform, static grid simulation there are no implicit or explicit HtoD or DtoH copies.
I also updated the CI machine file to Cuda 11.

Closing #334 

## PR Checklist

- [does not apply] Code passes cpplint
- [does not apply] New features are documented.
- [does not apply] Adds a test for any bugs fixed. Adds tests for new features.
- [does not apply]  Code is formatted
- [x] Changes are summarized in CHANGELOG.md

### CODE REVIEW

**Comment by pgrete** [comment]
Looks like this works.
Here's the profiling output https://pgrete.gitlab.io/-/parthenon/-/jobs/812120487/artifacts/build-cuda-perf-mpi/profile.txt which is automatically checked for `HtoD` or `DtoH` occurrences.
I'm now running the full test suite to make sure that the updated Cuda 11 environment also works for the other tests, but otherwise this is good to go.

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -79,14 +79,27 @@ parthenon-cuda-short:
...
+    # during loadbalance and refinement, but not for a static grid.
+    # Also delaying start as there are explicit copies during initialization, e.g.,
+    # when the Variable caches are created.
+    - nsys profile --delay=5 --duration=5 --stats=true example/advection/advection-example
+      -i ../tst/regression/test_suites/advection_performance/parthinput.advection_performance
+      parthenon/mesh/nx1=128 parthenon/mesh/nx2=128  parthenon/mesh/nx3=128
+      parthenon/meshblock/nx1=64 parthenon/meshblock/nx2=64 parthenon/meshblock/nx3=64
+      parthenon/time/nlim=1000 |& tee profile.txt
+    - test $(grep HtoD profile.txt |wc -l) == 0
+    - test $(grep DtoH profile.txt |wc -l) == 0
```
Comment (JoshuaSBrown) [inline .gitlab-ci.yml:96]: This is nice, I didn't realize bash had a test command like this.

**Review by JoshuaSBrown — APPROVED** [review]
Looks good to me.

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,7 @@
 - [[PR 311]](https://github.com/lanl/parthenon/pull/250) Bugfix::Restart. Fixed restart parallel bug and also restart bug for simulations with reflecting boundary conditions.  Added ability to write restart files with or without ghost cells by setting `ghost_zones` in the output block similar to other output formats.
 
 ### Added (new features/APIs/variables/...)
+- [[PR 336]](https://github.com/lanl/parthenon/pull/336) Automated testing now checks for extraneous HtoD or DtoH copies.
```
Comment (AndrewGaspar) [inline CHANGELOG.md]: We should probably have a separate section for "Infrastructure" or something...

**Review by AndrewGaspar — APPROVED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (pgrete) [reply CHANGELOG.md]: I think that's a good idea.
I now added a section on `Infrastructure (changes irrelevant to downstream codes)`

**Pull request merged**